File: appmonet-sdk/appmonet-sdk-mopub/src/main/java/com/monet/bidder/AdViewLoadedFactory.java
package com.monet.bidder;

import android.view.View;

import com.monet.bidder.bid.BidResponse;
import com.mopub.mobileads.MoPubView;

class AdViewLoadedFactory {
  View getAdView(SdkManager manager, View originalView, BidResponse bid, String adUnit) {
    final MoPubView mopubView = manager.getMopubAdView(adUnit);
    if (manager.currentActivity != null && manager.getFloatingAdPosition(adUnit) != null
        && mopubView != null) {
      FloatingAdView.Params params = new FloatingAdView.Params(manager, originalView, bid,
          mopubView.getAdWidth(), mopubView.getAdHeight(), adUnit);
      return new FloatingAdView(manager, params, originalView.getContext());
    } else if(manager.currentActivity!=null && manager.getFloatingAdPosition(adUnit) != null){
      FloatingAdView.Params params = new FloatingAdView.Params(manager, originalView, bid,
          null, null, adUnit);
      return new FloatingAdView(manager, params, originalView.getContext());
    }
    return originalView;
  }
}

File: appmonet-sdk/appmonet-sdk-mopub/src/main/java/com/monet/bidder/AdViewLoadedFactory.kt
package com.monet.bidder

import android.app.Activity
import android.view.View
import com.monet.bidder.FloatingAdView.Params
import com.monet.bidder.bid.BidResponse

internal class AdViewLoadedFactory {
  fun getAdView(
    activity: Activity?,
    manager: SdkManager,
    originalView: View,
    bid: BidResponse,
    adUnit: String
  ): View {
    val moPubView = manager.getMopubAdView(adUnit)
    if (activity != null && manager.getFloatingAdPosition(
            adUnit
        ) != null && moPubView != null
    ) {
      val params = Params(
          manager, originalView, bid,
          moPubView.getAdWidth(), moPubView.getAdHeight(), adUnit
      )
      return FloatingAdView(
          activity, manager, params, originalView.context
      )
    } else if (activity != null && manager.getFloatingAdPosition(
            adUnit
        ) != null
    ) {
      val params = Params(
          manager, originalView, bid,
          null, null, adUnit
      )
      return FloatingAdView(activity, manager, params, originalView.context)
    }
    return originalView
  }
}